compiler/check.go

```diff
@@ -0,0 +1,136 @@
+package compiler
+
```

Comment:
Can you add a short description at the top of this file that indicates what this file is for?
For example:
> This file implements a verifier for LLVM IR, to check for inconsistencies in the IR that are not checked by the LLVM verifier.